### PR TITLE
[senechome] implement more data points (3.0)

### PR DESCRIPTION
**Pull request by KorbinianP**
This is an Improvement PR. [My last one was based on 2.5.x branch and got canceled for a good reason](https://github.com/openhab/openhab-addons/pull/8858), here my rebase on the main branch.

### Background

To control the Wallbox for my electric car I need to know the current on each phase separately to avoid kicking the main fuse of my house. In Winter I could draw 32A on all three phases, the main Fuse has 35A per Phase. Leaves 3A for the rest of my house including my Heating Pump. This is by far too less, I need to reduce the charging speed. But if the sun is shining, my PV will give me additional power that comes on top of the Grid, I can increase charging speed. And while I was on it I added some more values that could be interesting.

### Summary of changes

I added these channels:

- SenecGridPowerPh1: Power draw from grid on Phase 1
- SenecGridPowerPh2: Power draw from grid on Phase 2
- SenecGridPowerPh3: Power draw from grid on Phase 3
- SenecGridCurrentPh1: Current draw from grid on Phase 1
- SenecGridCurrentPh2: Current draw from grid on Phase 2
- SenecGridCurrentPh3: Current draw from grid on Phase 3
- SenecGridVoltagePh1: Voltage Level on Phase 1
- SenecGridVoltagePh2: Voltage Level on Phase 2
- SenecGridVoltagePh3: Voltage Level on Phase 3
- SenecGridFrequency: Grid Frequency
- SenecBatteryStateValue: Value describing current action of battery (e.g. 14)
- SenecBatteryVoltage: Battery Voltage
- SenecLiveBatCharge: Live Bat Charge
- SenecLiveBatDischarge: Live Bat Discharge
- SenecLiveGridImport: Live Grid Import
- SenecLiveGridExport: Live Grid Export

### Download

2.5: [org.openhab.binding.senechome-2.5.11-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/5477415/org.openhab.binding.senechome-2.5.11-SNAPSHOT.zip)
3.0: [org.openhab.binding.senechome-3.0.0-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/5477412/org.openhab.binding.senechome-3.0.0-SNAPSHOT.zip)

### CODE REVIEW

**Comment by KorbinianP** [comment]
Hello all,
I am done with my desired changes.
Cheers

File: bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java [hunk 1]
```diff
@@ -156,6 +160,75 @@ private void refresh() {
...
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_GRID_FREQUENCY);
+            updateState(channelGridFrequency.getUID(),
+                    new QuantityType<Frequency>(
+                            getSenecValue(response.grid.currentGridFrequency).setScale(2, RoundingMode.HALF_UP),
+                            SmartHomeUnits.HERTZ));
+
+            Channel channelBatteryStateValue = getThing()
+                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_BATTERY_STATE_VALUE);
+            updateState(channelBatteryStateValue.getUID(),
+                    new DecimalType(getSenecValue(response.energy.batteryState).intValue()));
```
Comment (cpmeister) [inline bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java:230]: You specified `Number:Dimensionless` so you should use a QuantityType here.

**Review by cpmeister — CHANGES_REQUESTED** [review]
Looks good, just one small comment.

File: bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java (hunk 1 above)
Comment (KorbinianP) [reply bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java:230]: Hello, but this is just a decimal without any Type. In the end it is a value of an enum. I don't find any good SmartHomeUnits Type for that. Do you have a proposal what I could give as second param if I go for `QuantityType<Dimensionless>`?

```
updateState(channelBatteryStateValue.getUID(),
    new QuantityType<Dimensionless>(getSenecValue(response.energy.batteryState).intValue()),
    SmartHomeUnits.**????**);
```

Comment (cpmeister) [reply bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java:230]: SmartHomeUnits.ONE or Units.ONE

Comment (cpmeister) [reply bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java:230]: Although if it is really an enum then you shouldn't specify it as `Number:Dimensionless` in the xml and instead just specify it as a `Number` instead.

Comment (KorbinianP) [reply bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java]: Done the change 👍

File: bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java [hunk 2]
```diff
@@ -228,7 +227,7 @@ private void refresh() {
             Channel channelBatteryStateValue = getThing()
                     .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_BATTERY_STATE_VALUE);
             updateState(channelBatteryStateValue.getUID(),
-                    new DecimalType(getSenecValue(response.energy.batteryState).intValue()));
+                    new QuantityType<>(getSenecValue(response.energy.batteryState).intValue(), SmartHomeUnits.ONE));
```
Comment (cpmeister) [inline bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java]: You switched the item-type to Number so you need to change this back and use DecimalType again.

**Review by cpmeister — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java (hunk 2 above)
Comment (KorbinianP) [reply bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java]: OK, done that. JARs in the opener are updated again 👍

**Review by cpmeister — APPROVED** [review]
LGTM

**Comment by KorbinianP** [comment]
Hello @cpmeister ,
thank you for the review and merge! Is there a way to backport this officially to 2.5? The branch would be ready.
Cheers

**Comment by cpmeister** [comment]
Not really sure if we are still going to perform any more releases for 2.5.x. 
Feel free to make another PR against the 2.5 branch with your changes though.

**Comment by kaikreuzer** [comment]
@KorbinianP 2.5.x should really only be patch releases from now on - so bugs can be backported, but features shouldn't really.

**Comment by KorbinianP** [comment]
Ok, then I will provide the Jar above also in the OpenHab forum the next days. For all that don't want to update to 3.0 on the first days but still want new features :-)
Cheers,

**Pull request merged**